flink/src/main/scala/com/snowplowanalytics/snowplow/event/recovery/RecoveryJob.scala

```diff
@@ -0,0 +1,141 @@
+import org.apache.flink.connector.aws.config.AWSConfigConstants
+import org.apache.flink.api.common.serialization.SimpleStringEncoder
+import org.apache.flink.api.common.serialization.SimpleStringSchema
+
+object RecoveryJob extends RecoveryJob
+
+trait RecoveryJob {
+
+  /** Flink job running the event recovery process on AWS. It will:
+    *   - read the input data from an S3 location
+    *   - decode the bad row jsons
+    *   - mutate the collector payloads contained in the concerned bad rows according to the specified recovery
+    *     scenarios
+    *   - write out the fixed payloads to Kinesis
+    *   - write failed recoveries to an S3 location
+    *   - write unrecoverable bad rows to an S3 location
+    * @param input
+    *   S3 location to read the bad rows from
+    * @param output
+    *   Kinesis stream to write the fixed collector payloads to
+    * @param failedOutput
+    *   S3 location to write the failed recoveries
+    * @param unrecoverableOutput
+    *   S3 location to write unrecoverble bad rows
+    * @param cfg
+    *   configuration object containing mappings and recovery flow configurations
+    * @param kinesisConfig
+    *   configuration for Kinesis sink
+    */
+  def run(
+    env: StreamExecutionEnvironment,
+    input: String,
+    output: String,
+    failedOutput: String,
+    unrecoverableOutput: String,
+    cfg: Config,
+    kinesisConfig: KinesisConfig
+  ): Unit = {
+    val properties = new Properties()
+    properties.put(AWSConfigConstants.AWS_REGION, kinesisConfig.region)
+    kinesisConfig.credentials.foreach { case Credentials(accessKey, secretKey) =>
+      properties.put(AWSConfigConstants.AWS_ACCESS_KEY_ID, accessKey)
+      properties.put(AWSConfigConstants.AWS_SECRET_ACCESS_KEY, secretKey)
+    }
+    kinesisConfig.endpoint.foreach { endpoint =>
+      properties.put(AWSConfigConstants.AWS_ENDPOINT, endpoint)
+    }
+
+    val kinesisBuilder = KinesisStreamsSink
+      .builder()
+      .setKinesisClientProperties(properties)
+      .setSerializationSchema(new BytesSerializationSchema())
+      .setPartitionKeyGenerator(new Partitioner())
+      .setStreamName(output)
+      .setFailOnError(false)
+
+    kinesisConfig.batchSize.foreach(kinesisBuilder.setMaxBatchSize)
+    kinesisConfig.maxBuffered.foreach(kinesisBuilder.setMaxBufferedRequests)
+
+    val kinesis = kinesisBuilder.build()
+
+    val failed        = OutputTag[RecoveryError]("failed")
+    val unrecoverable = OutputTag[RecoveryError]("unrecoverable")
+
+    val recovering = load(env, input).map(execute(cfg)(_)).process(new SplitByStatus(failed, unrecoverable))
+
+    recovering.sinkTo(kinesis)
+
+    sinkErr(recovering, failed, failedOutput)
+    sinkErr(recovering, unrecoverable, unrecoverableOutput)
+
+    env.enableCheckpointing(1000)
+    env.execute("Event recovery job started.")
+    ()
+  }
+
+  def load(env: StreamExecutionEnvironment, input: String): DataStream[String] =
+    env.readTextFile(input)
```

Comment:
I see the lib uses `UTF-8` by default but given that `SimpleStringEncoder` below explicitly uses `UTF-8` (which is also the default in case we don't provide), I recommend we stay consistent & explicitly use `readTextFile(input, "UTF-8")`